quiz/equal-iv_en.md

```diff
@@ -1,4 +1,31 @@
+> This awesome solution is created by [@abhijeet-saxena](https://github.com/abhijeet-saxena)
 
-There is no solution yet.
+## Problem
 
-Would you like to [contribute to the solution](https://github.com/BFEdev/BFE.dev-solutions/blob/main/quiz/equal-iv_en.md)? [Contribute guideline](https://github.com/BFEdev/BFE.dev-solutions#how-to-contribute)
+```js
+console.log([0] == "");
+console.log([0] == 0);
+```
+
+## Solution
+
+```
+false
+true
+```
+
+## Explanation
+
+The [equality (==) operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Equality) checks whether its two operands are equal, and unlike the strict equality operator ===, it attempts to convert and compare operands that are of different types. We have,
+
+```
+[0].toString() // "0"
+0..toString() // "0"
+```
+
+In this example, we are comparing different data types which means `==` will convert the operands to strings, and then compare. So, the first comparison evaluates to `false`, and the second one is `true`
```

Comment:
> In this example, we are comparing different data types which means `==` will convert the operands to strings, and then compare

This doesn't seem right according to ECMAScript spec

Reply:
Yup. I think I misunderstood the comparison and oversimplified it a lot. I've updated it with more details.